src/st/no0/e_diplocephalus.c

```diff
@@ -1,10 +1,71 @@
 // SPDX-License-Identifier: AGPL-3.0-or-later
 #include "no0.h"
 
-INCLUDE_ASM("st/no0/nonmatchings/e_diplocephalus", func_us_801CF4A8);
+// func_us_801CF4A8
+extern s32 D_us_80181CC4[];    // velocityX
+extern s32 D_us_80181CE8[][8]; // velocityY
+extern s16 D_us_80181D54[];    // sensors_ground
 
-extern s16 D_us_80181D54[]; // sensors_ground
-extern u8 D_us_80181D83[];  // anim frames
+// func_us_801CF910
+extern s32 D_us_80181CD4[]; // velocityX
+extern s16 D_us_80181D64[]; // sensors_ground
+extern u8 D_us_80181DAC[];  // anims
+extern u8 D_us_80181DB4[];  // anims
+
+// Foot
+extern u8 D_us_80181D83[]; // anim frames
+
+// Some sort of death entity
+// Perhaps the parts that go flying on death
+void func_us_801CF4A8(Entity* self) {
+    Entity* newEntity;
+
+    if (self->ext.diplocephalus.parent->entityId != 0x21) {
```

Comment:
enum for the ID?

Reply:
This is one of the not yet decompiled functions `func_us_801CFBE8` so I was going to update all these then but I can add an enum now